[2021-12-07 16:11:19,851] {taskinstance.py:1035} INFO - Dependencies all met for <TaskInstance: scraping_daft_laois.persist_data scheduled__2021-12-06T02:45:00+00:00 [queued]>
[2021-12-07 16:11:19,903] {taskinstance.py:1035} INFO - Dependencies all met for <TaskInstance: scraping_daft_laois.persist_data scheduled__2021-12-06T02:45:00+00:00 [queued]>
[2021-12-07 16:11:19,904] {taskinstance.py:1241} INFO - 
--------------------------------------------------------------------------------
[2021-12-07 16:11:19,906] {taskinstance.py:1242} INFO - Starting attempt 2 of 3
[2021-12-07 16:11:19,907] {taskinstance.py:1243} INFO - 
--------------------------------------------------------------------------------
[2021-12-07 16:11:19,929] {taskinstance.py:1262} INFO - Executing <Task(SparkSubmitOperator): persist_data> on 2021-12-06 02:45:00+00:00
[2021-12-07 16:11:19,944] {standard_task_runner.py:76} INFO - Running: ['***', 'tasks', 'run', 'scraping_daft_laois', 'persist_data', 'scheduled__2021-12-06T02:45:00+00:00', '--job-id', '320', '--raw', '--subdir', 'DAGS_FOLDER/scraping_daft_laois.py', '--cfg-path', '/tmp/tmpj9lfluis', '--error-file', '/tmp/tmp1zep5o3s']
[2021-12-07 16:11:19,946] {standard_task_runner.py:77} INFO - Job 320: Subtask persist_data
[2021-12-07 16:11:19,939] {standard_task_runner.py:52} INFO - Started process 45571 to run task
[2021-12-07 16:11:20,073] {logging_mixin.py:109} INFO - Running <TaskInstance: scraping_daft_laois.persist_data scheduled__2021-12-06T02:45:00+00:00 [running]> on host 7a06f7031264
[2021-12-07 16:11:20,298] {taskinstance.py:1429} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=***@example.com
AIRFLOW_CTX_DAG_OWNER=clemoni
AIRFLOW_CTX_DAG_ID=scraping_daft_laois
AIRFLOW_CTX_TASK_ID=persist_data
AIRFLOW_CTX_EXECUTION_DATE=2021-12-06T02:45:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2021-12-06T02:45:00+00:00
[2021-12-07 16:11:20,346] {base.py:79} INFO - Using connection to: id: spark_default. Host: spark://spark:7077, Port: None, Schema: , Login: ***, Password: ***, extra: {}
[2021-12-07 16:11:20,354] {spark_submit.py:362} INFO - Spark-Submit cmd: spark-submit --master spark://spark:7077 --conf spark.master=spark://spark:7077 --name persist data --verbose /opt/***/jobs/persist_db.py laois
[2021-12-07 16:11:22,615] {spark_submit.py:523} INFO - Using properties file: null
[2021-12-07 16:11:22,780] {spark_submit.py:523} INFO - WARNING: An illegal reflective access operation has occurred
[2021-12-07 16:11:22,781] {spark_submit.py:523} INFO - WARNING: Illegal reflective access by org.apache.spark.unsafe.Platform (file:/home/***/.local/lib/python3.6/site-packages/pyspark/jars/spark-unsafe_2.12-3.2.0.jar) to constructor java.nio.DirectByteBuffer(long,int)
[2021-12-07 16:11:22,782] {spark_submit.py:523} INFO - WARNING: Please consider reporting this to the maintainers of org.apache.spark.unsafe.Platform
[2021-12-07 16:11:22,783] {spark_submit.py:523} INFO - WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
[2021-12-07 16:11:22,784] {spark_submit.py:523} INFO - WARNING: All illegal access operations will be denied in a future release
[2021-12-07 16:11:22,872] {spark_submit.py:523} INFO - Parsed arguments:
[2021-12-07 16:11:22,873] {spark_submit.py:523} INFO - master                  spark://spark:7077
[2021-12-07 16:11:22,874] {spark_submit.py:523} INFO - deployMode              null
[2021-12-07 16:11:22,876] {spark_submit.py:523} INFO - executorMemory          null
[2021-12-07 16:11:22,877] {spark_submit.py:523} INFO - executorCores           null
[2021-12-07 16:11:22,879] {spark_submit.py:523} INFO - totalExecutorCores      null
[2021-12-07 16:11:22,880] {spark_submit.py:523} INFO - propertiesFile          null
[2021-12-07 16:11:22,881] {spark_submit.py:523} INFO - driverMemory            null
[2021-12-07 16:11:22,883] {spark_submit.py:523} INFO - driverCores             null
[2021-12-07 16:11:22,884] {spark_submit.py:523} INFO - driverExtraClassPath    null
[2021-12-07 16:11:22,885] {spark_submit.py:523} INFO - driverExtraLibraryPath  null
[2021-12-07 16:11:22,887] {spark_submit.py:523} INFO - driverExtraJavaOptions  null
[2021-12-07 16:11:22,888] {spark_submit.py:523} INFO - supervise               false
[2021-12-07 16:11:22,889] {spark_submit.py:523} INFO - queue                   null
[2021-12-07 16:11:22,890] {spark_submit.py:523} INFO - numExecutors            null
[2021-12-07 16:11:22,891] {spark_submit.py:523} INFO - files                   null
[2021-12-07 16:11:22,893] {spark_submit.py:523} INFO - pyFiles                 null
[2021-12-07 16:11:22,894] {spark_submit.py:523} INFO - archives                null
[2021-12-07 16:11:22,895] {spark_submit.py:523} INFO - mainClass               null
[2021-12-07 16:11:22,896] {spark_submit.py:523} INFO - primaryResource         file:/opt/***/jobs/persist_db.py
[2021-12-07 16:11:22,897] {spark_submit.py:523} INFO - name                    persist data
[2021-12-07 16:11:22,898] {spark_submit.py:523} INFO - childArgs               [laois]
[2021-12-07 16:11:22,899] {spark_submit.py:523} INFO - jars                    null
[2021-12-07 16:11:22,900] {spark_submit.py:523} INFO - packages                null
[2021-12-07 16:11:22,901] {spark_submit.py:523} INFO - packagesExclusions      null
[2021-12-07 16:11:22,902] {spark_submit.py:523} INFO - repositories            null
[2021-12-07 16:11:22,904] {spark_submit.py:523} INFO - verbose                 true
[2021-12-07 16:11:22,905] {spark_submit.py:523} INFO - 
[2021-12-07 16:11:22,906] {spark_submit.py:523} INFO - Spark properties used, including those specified through
[2021-12-07 16:11:22,907] {spark_submit.py:523} INFO - --conf and those from the properties file null:
[2021-12-07 16:11:22,908] {spark_submit.py:523} INFO - (spark.master,spark://spark:7077)
[2021-12-07 16:11:22,910] {spark_submit.py:523} INFO - 
[2021-12-07 16:11:22,911] {spark_submit.py:523} INFO - 
[2021-12-07 16:11:23,318] {spark_submit.py:523} INFO - Main class:
[2021-12-07 16:11:23,320] {spark_submit.py:523} INFO - org.apache.spark.deploy.PythonRunner
[2021-12-07 16:11:23,321] {spark_submit.py:523} INFO - Arguments:
[2021-12-07 16:11:23,322] {spark_submit.py:523} INFO - file:/opt/***/jobs/persist_db.py
[2021-12-07 16:11:23,323] {spark_submit.py:523} INFO - null
[2021-12-07 16:11:23,324] {spark_submit.py:523} INFO - laois
[2021-12-07 16:11:23,325] {spark_submit.py:523} INFO - --verbose
[2021-12-07 16:11:23,326] {spark_submit.py:523} INFO - Spark config:
[2021-12-07 16:11:23,327] {spark_submit.py:523} INFO - (spark.master,spark://spark:7077)
[2021-12-07 16:11:23,329] {spark_submit.py:523} INFO - (spark.app.name,persist data)
[2021-12-07 16:11:23,330] {spark_submit.py:523} INFO - (spark.submit.pyFiles,)
[2021-12-07 16:11:23,331] {spark_submit.py:523} INFO - (spark.submit.deployMode,client)
[2021-12-07 16:11:23,332] {spark_submit.py:523} INFO - Classpath elements:
[2021-12-07 16:11:23,333] {spark_submit.py:523} INFO - 
[2021-12-07 16:11:23,334] {spark_submit.py:523} INFO - 
[2021-12-07 16:11:23,336] {spark_submit.py:523} INFO - 
[2021-12-07 16:11:25,101] {spark_submit.py:523} INFO - Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
[2021-12-07 16:11:25,123] {spark_submit.py:523} INFO - 21/12/07 16:11:25 INFO SparkContext: Running Spark version 3.2.0
[2021-12-07 16:11:25,266] {spark_submit.py:523} INFO - 21/12/07 16:11:25 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2021-12-07 16:11:25,551] {spark_submit.py:523} INFO - 21/12/07 16:11:25 INFO ResourceUtils: ==============================================================
[2021-12-07 16:11:25,552] {spark_submit.py:523} INFO - 21/12/07 16:11:25 INFO ResourceUtils: No custom resources configured for spark.driver.
[2021-12-07 16:11:25,553] {spark_submit.py:523} INFO - 21/12/07 16:11:25 INFO ResourceUtils: ==============================================================
[2021-12-07 16:11:25,554] {spark_submit.py:523} INFO - 21/12/07 16:11:25 INFO SparkContext: Submitted application: Persist DB
[2021-12-07 16:11:25,601] {spark_submit.py:523} INFO - 21/12/07 16:11:25 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2021-12-07 16:11:25,622] {spark_submit.py:523} INFO - 21/12/07 16:11:25 INFO ResourceProfile: Limiting resource is cpu
[2021-12-07 16:11:25,624] {spark_submit.py:523} INFO - 21/12/07 16:11:25 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2021-12-07 16:11:25,739] {spark_submit.py:523} INFO - 21/12/07 16:11:25 INFO SecurityManager: Changing view acls to: ***
[2021-12-07 16:11:25,740] {spark_submit.py:523} INFO - 21/12/07 16:11:25 INFO SecurityManager: Changing modify acls to: ***
[2021-12-07 16:11:25,741] {spark_submit.py:523} INFO - 21/12/07 16:11:25 INFO SecurityManager: Changing view acls groups to:
[2021-12-07 16:11:25,742] {spark_submit.py:523} INFO - 21/12/07 16:11:25 INFO SecurityManager: Changing modify acls groups to:
[2021-12-07 16:11:25,743] {spark_submit.py:523} INFO - 21/12/07 16:11:25 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(***); groups with view permissions: Set(); users  with modify permissions: Set(***); groups with modify permissions: Set()
[2021-12-07 16:11:26,275] {spark_submit.py:523} INFO - 21/12/07 16:11:26 INFO Utils: Successfully started service 'sparkDriver' on port 42129.
[2021-12-07 16:11:26,345] {spark_submit.py:523} INFO - 21/12/07 16:11:26 INFO SparkEnv: Registering MapOutputTracker
[2021-12-07 16:11:26,409] {spark_submit.py:523} INFO - 21/12/07 16:11:26 INFO SparkEnv: Registering BlockManagerMaster
[2021-12-07 16:11:26,453] {spark_submit.py:523} INFO - 21/12/07 16:11:26 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2021-12-07 16:11:26,455] {spark_submit.py:523} INFO - 21/12/07 16:11:26 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2021-12-07 16:11:26,465] {spark_submit.py:523} INFO - 21/12/07 16:11:26 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2021-12-07 16:11:26,510] {spark_submit.py:523} INFO - 21/12/07 16:11:26 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-6fa8b94a-ae52-418a-9387-faad24e193fd
[2021-12-07 16:11:26,547] {spark_submit.py:523} INFO - 21/12/07 16:11:26 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2021-12-07 16:11:26,580] {spark_submit.py:523} INFO - 21/12/07 16:11:26 INFO SparkEnv: Registering OutputCommitCoordinator
[2021-12-07 16:11:27,125] {spark_submit.py:523} INFO - 21/12/07 16:11:27 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2021-12-07 16:11:27,153] {spark_submit.py:523} INFO - 21/12/07 16:11:27 INFO Utils: Successfully started service 'SparkUI' on port 4041.
[2021-12-07 16:11:27,298] {spark_submit.py:523} INFO - 21/12/07 16:11:27 INFO SparkUI: Bound SparkUI to 0.0.0.0, and started at http://7a06f7031264:4041
[2021-12-07 16:11:27,641] {spark_submit.py:523} INFO - 21/12/07 16:11:27 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark:7077...
[2021-12-07 16:11:27,759] {spark_submit.py:523} INFO - 21/12/07 16:11:27 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark:7077
[2021-12-07 16:11:27,760] {spark_submit.py:523} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult:
[2021-12-07 16:11:27,761] {spark_submit.py:523} INFO - at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:301)
[2021-12-07 16:11:27,762] {spark_submit.py:523} INFO - at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2021-12-07 16:11:27,763] {spark_submit.py:523} INFO - at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:101)
[2021-12-07 16:11:27,764] {spark_submit.py:523} INFO - at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:109)
[2021-12-07 16:11:27,765] {spark_submit.py:523} INFO - at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:107)
[2021-12-07 16:11:27,767] {spark_submit.py:523} INFO - at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[2021-12-07 16:11:27,768] {spark_submit.py:523} INFO - at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2021-12-07 16:11:27,769] {spark_submit.py:523} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2021-12-07 16:11:27,770] {spark_submit.py:523} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2021-12-07 16:11:27,771] {spark_submit.py:523} INFO - at java.base/java.lang.Thread.run(Thread.java:829)
[2021-12-07 16:11:27,772] {spark_submit.py:523} INFO - Caused by: java.io.IOException: Failed to connect to spark:7077
[2021-12-07 16:11:27,773] {spark_submit.py:523} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:288)
[2021-12-07 16:11:27,773] {spark_submit.py:523} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:218)
[2021-12-07 16:11:27,774] {spark_submit.py:523} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
[2021-12-07 16:11:27,775] {spark_submit.py:523} INFO - at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
[2021-12-07 16:11:27,776] {spark_submit.py:523} INFO - at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
[2021-12-07 16:11:27,777] {spark_submit.py:523} INFO - at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
[2021-12-07 16:11:27,778] {spark_submit.py:523} INFO - ... 4 more
[2021-12-07 16:11:27,779] {spark_submit.py:523} INFO - Caused by: java.net.UnknownHostException: spark
[2021-12-07 16:11:27,780] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:797)
[2021-12-07 16:11:27,781] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1509)
[2021-12-07 16:11:27,782] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1368)
[2021-12-07 16:11:27,783] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1302)
[2021-12-07 16:11:27,784] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress.getByName(InetAddress.java:1252)
[2021-12-07 16:11:27,785] {spark_submit.py:523} INFO - at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156)
[2021-12-07 16:11:27,786] {spark_submit.py:523} INFO - at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153)
[2021-12-07 16:11:27,787] {spark_submit.py:523} INFO - at java.base/java.security.AccessController.doPrivileged(Native Method)
[2021-12-07 16:11:27,788] {spark_submit.py:523} INFO - at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153)
[2021-12-07 16:11:27,789] {spark_submit.py:523} INFO - at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41)
[2021-12-07 16:11:27,790] {spark_submit.py:523} INFO - at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61)
[2021-12-07 16:11:27,790] {spark_submit.py:523} INFO - at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53)
[2021-12-07 16:11:27,791] {spark_submit.py:523} INFO - at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55)
[2021-12-07 16:11:27,792] {spark_submit.py:523} INFO - at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31)
[2021-12-07 16:11:27,793] {spark_submit.py:523} INFO - at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106)
[2021-12-07 16:11:27,794] {spark_submit.py:523} INFO - at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:206)
[2021-12-07 16:11:27,795] {spark_submit.py:523} INFO - at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:46)
[2021-12-07 16:11:27,796] {spark_submit.py:523} INFO - at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:180)
[2021-12-07 16:11:27,797] {spark_submit.py:523} INFO - at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:166)
[2021-12-07 16:11:27,798] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
[2021-12-07 16:11:27,799] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:552)
[2021-12-07 16:11:27,800] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491)
[2021-12-07 16:11:27,801] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616)
[2021-12-07 16:11:27,802] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:605)
[2021-12-07 16:11:27,803] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
[2021-12-07 16:11:27,804] {spark_submit.py:523} INFO - at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
[2021-12-07 16:11:27,808] {spark_submit.py:523} INFO - at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:1008)
[2021-12-07 16:11:27,811] {spark_submit.py:523} INFO - at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:516)
[2021-12-07 16:11:27,812] {spark_submit.py:523} INFO - at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429)
[2021-12-07 16:11:27,813] {spark_submit.py:523} INFO - at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486)
[2021-12-07 16:11:27,814] {spark_submit.py:523} INFO - at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
[2021-12-07 16:11:27,815] {spark_submit.py:523} INFO - at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
[2021-12-07 16:11:27,817] {spark_submit.py:523} INFO - at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
[2021-12-07 16:11:27,817] {spark_submit.py:523} INFO - at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
[2021-12-07 16:11:27,818] {spark_submit.py:523} INFO - at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[2021-12-07 16:11:27,819] {spark_submit.py:523} INFO - at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[2021-12-07 16:11:27,820] {spark_submit.py:523} INFO - ... 1 more
[2021-12-07 16:11:47,601] {spark_submit.py:523} INFO - 21/12/07 16:11:47 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark:7077...
[2021-12-07 16:11:47,649] {spark_submit.py:523} INFO - 21/12/07 16:11:47 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark:7077
[2021-12-07 16:11:47,653] {spark_submit.py:523} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult:
[2021-12-07 16:11:47,656] {spark_submit.py:523} INFO - at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:301)
[2021-12-07 16:11:47,659] {spark_submit.py:523} INFO - at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2021-12-07 16:11:47,662] {spark_submit.py:523} INFO - at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:101)
[2021-12-07 16:11:47,668] {spark_submit.py:523} INFO - at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:109)
[2021-12-07 16:11:47,669] {spark_submit.py:523} INFO - at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:107)
[2021-12-07 16:11:47,671] {spark_submit.py:523} INFO - at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[2021-12-07 16:11:47,672] {spark_submit.py:523} INFO - at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2021-12-07 16:11:47,674] {spark_submit.py:523} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2021-12-07 16:11:47,678] {spark_submit.py:523} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2021-12-07 16:11:47,681] {spark_submit.py:523} INFO - at java.base/java.lang.Thread.run(Thread.java:829)
[2021-12-07 16:11:47,683] {spark_submit.py:523} INFO - Caused by: java.io.IOException: Failed to connect to spark:7077
[2021-12-07 16:11:47,685] {spark_submit.py:523} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:288)
[2021-12-07 16:11:47,688] {spark_submit.py:523} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:218)
[2021-12-07 16:11:47,690] {spark_submit.py:523} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
[2021-12-07 16:11:47,693] {spark_submit.py:523} INFO - at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
[2021-12-07 16:11:47,697] {spark_submit.py:523} INFO - at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
[2021-12-07 16:11:47,699] {spark_submit.py:523} INFO - at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
[2021-12-07 16:11:47,701] {spark_submit.py:523} INFO - ... 4 more
[2021-12-07 16:11:47,702] {spark_submit.py:523} INFO - Caused by: java.net.UnknownHostException: spark
[2021-12-07 16:11:47,704] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:797)
[2021-12-07 16:11:47,706] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1509)
[2021-12-07 16:11:47,707] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1368)
[2021-12-07 16:11:47,708] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1302)
[2021-12-07 16:11:47,710] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress.getByName(InetAddress.java:1252)
[2021-12-07 16:11:47,711] {spark_submit.py:523} INFO - at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156)
[2021-12-07 16:11:47,713] {spark_submit.py:523} INFO - at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153)
[2021-12-07 16:11:47,715] {spark_submit.py:523} INFO - at java.base/java.security.AccessController.doPrivileged(Native Method)
[2021-12-07 16:11:47,716] {spark_submit.py:523} INFO - at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153)
[2021-12-07 16:11:47,718] {spark_submit.py:523} INFO - at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41)
[2021-12-07 16:11:47,719] {spark_submit.py:523} INFO - at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61)
[2021-12-07 16:11:47,722] {spark_submit.py:523} INFO - at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53)
[2021-12-07 16:11:47,724] {spark_submit.py:523} INFO - at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55)
[2021-12-07 16:11:47,727] {spark_submit.py:523} INFO - at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31)
[2021-12-07 16:11:47,729] {spark_submit.py:523} INFO - at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106)
[2021-12-07 16:11:47,731] {spark_submit.py:523} INFO - at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:206)
[2021-12-07 16:11:47,733] {spark_submit.py:523} INFO - at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:46)
[2021-12-07 16:11:47,735] {spark_submit.py:523} INFO - at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:180)
[2021-12-07 16:11:47,737] {spark_submit.py:523} INFO - at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:166)
[2021-12-07 16:11:47,738] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
[2021-12-07 16:11:47,742] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:552)
[2021-12-07 16:11:47,743] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491)
[2021-12-07 16:11:47,745] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616)
[2021-12-07 16:11:47,746] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:605)
[2021-12-07 16:11:47,748] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
[2021-12-07 16:11:47,749] {spark_submit.py:523} INFO - at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
[2021-12-07 16:11:47,751] {spark_submit.py:523} INFO - at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:1008)
[2021-12-07 16:11:47,752] {spark_submit.py:523} INFO - at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:516)
[2021-12-07 16:11:47,754] {spark_submit.py:523} INFO - at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429)
[2021-12-07 16:11:47,755] {spark_submit.py:523} INFO - at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486)
[2021-12-07 16:11:47,757] {spark_submit.py:523} INFO - at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
[2021-12-07 16:11:47,758] {spark_submit.py:523} INFO - at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
[2021-12-07 16:11:47,760] {spark_submit.py:523} INFO - at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
[2021-12-07 16:11:47,761] {spark_submit.py:523} INFO - at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
[2021-12-07 16:11:47,762] {spark_submit.py:523} INFO - at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[2021-12-07 16:11:47,764] {spark_submit.py:523} INFO - at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[2021-12-07 16:11:47,765] {spark_submit.py:523} INFO - ... 1 more
[2021-12-07 16:12:07,567] {spark_submit.py:523} INFO - 21/12/07 16:12:07 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark:7077...
[2021-12-07 16:12:07,584] {spark_submit.py:523} INFO - 21/12/07 16:12:07 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark:7077
[2021-12-07 16:12:07,585] {spark_submit.py:523} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult:
[2021-12-07 16:12:07,586] {spark_submit.py:523} INFO - at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:301)
[2021-12-07 16:12:07,587] {spark_submit.py:523} INFO - at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2021-12-07 16:12:07,588] {spark_submit.py:523} INFO - at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:101)
[2021-12-07 16:12:07,588] {spark_submit.py:523} INFO - at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:109)
[2021-12-07 16:12:07,589] {spark_submit.py:523} INFO - at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:107)
[2021-12-07 16:12:07,590] {spark_submit.py:523} INFO - at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[2021-12-07 16:12:07,590] {spark_submit.py:523} INFO - at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2021-12-07 16:12:07,591] {spark_submit.py:523} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2021-12-07 16:12:07,592] {spark_submit.py:523} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2021-12-07 16:12:07,592] {spark_submit.py:523} INFO - at java.base/java.lang.Thread.run(Thread.java:829)
[2021-12-07 16:12:07,594] {spark_submit.py:523} INFO - Caused by: java.io.IOException: Failed to connect to spark:7077
[2021-12-07 16:12:07,595] {spark_submit.py:523} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:288)
[2021-12-07 16:12:07,596] {spark_submit.py:523} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:218)
[2021-12-07 16:12:07,596] {spark_submit.py:523} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
[2021-12-07 16:12:07,597] {spark_submit.py:523} INFO - at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
[2021-12-07 16:12:07,598] {spark_submit.py:523} INFO - at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
[2021-12-07 16:12:07,598] {spark_submit.py:523} INFO - at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
[2021-12-07 16:12:07,599] {spark_submit.py:523} INFO - ... 4 more
[2021-12-07 16:12:07,600] {spark_submit.py:523} INFO - Caused by: java.net.UnknownHostException: spark
[2021-12-07 16:12:07,601] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:797)
[2021-12-07 16:12:07,601] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1509)
[2021-12-07 16:12:07,602] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1368)
[2021-12-07 16:12:07,603] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1302)
[2021-12-07 16:12:07,603] {spark_submit.py:523} INFO - at java.base/java.net.InetAddress.getByName(InetAddress.java:1252)
[2021-12-07 16:12:07,604] {spark_submit.py:523} INFO - at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:156)
[2021-12-07 16:12:07,605] {spark_submit.py:523} INFO - at io.netty.util.internal.SocketUtils$8.run(SocketUtils.java:153)
[2021-12-07 16:12:07,605] {spark_submit.py:523} INFO - at java.base/java.security.AccessController.doPrivileged(Native Method)
[2021-12-07 16:12:07,606] {spark_submit.py:523} INFO - at io.netty.util.internal.SocketUtils.addressByName(SocketUtils.java:153)
[2021-12-07 16:12:07,607] {spark_submit.py:523} INFO - at io.netty.resolver.DefaultNameResolver.doResolve(DefaultNameResolver.java:41)
[2021-12-07 16:12:07,608] {spark_submit.py:523} INFO - at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:61)
[2021-12-07 16:12:07,608] {spark_submit.py:523} INFO - at io.netty.resolver.SimpleNameResolver.resolve(SimpleNameResolver.java:53)
[2021-12-07 16:12:07,609] {spark_submit.py:523} INFO - at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:55)
[2021-12-07 16:12:07,610] {spark_submit.py:523} INFO - at io.netty.resolver.InetSocketAddressResolver.doResolve(InetSocketAddressResolver.java:31)
[2021-12-07 16:12:07,611] {spark_submit.py:523} INFO - at io.netty.resolver.AbstractAddressResolver.resolve(AbstractAddressResolver.java:106)
[2021-12-07 16:12:07,612] {spark_submit.py:523} INFO - at io.netty.bootstrap.Bootstrap.doResolveAndConnect0(Bootstrap.java:206)
[2021-12-07 16:12:07,612] {spark_submit.py:523} INFO - at io.netty.bootstrap.Bootstrap.access$000(Bootstrap.java:46)
[2021-12-07 16:12:07,613] {spark_submit.py:523} INFO - at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:180)
[2021-12-07 16:12:07,614] {spark_submit.py:523} INFO - at io.netty.bootstrap.Bootstrap$1.operationComplete(Bootstrap.java:166)
[2021-12-07 16:12:07,615] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
[2021-12-07 16:12:07,615] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:552)
[2021-12-07 16:12:07,616] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491)
[2021-12-07 16:12:07,617] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616)
[2021-12-07 16:12:07,618] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:605)
[2021-12-07 16:12:07,619] {spark_submit.py:523} INFO - at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
[2021-12-07 16:12:07,620] {spark_submit.py:523} INFO - at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
[2021-12-07 16:12:07,621] {spark_submit.py:523} INFO - at io.netty.channel.AbstractChannel$AbstractUnsafe.safeSetSuccess(AbstractChannel.java:1008)
[2021-12-07 16:12:07,621] {spark_submit.py:523} INFO - at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:516)
[2021-12-07 16:12:07,622] {spark_submit.py:523} INFO - at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:429)
[2021-12-07 16:12:07,623] {spark_submit.py:523} INFO - at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:486)
[2021-12-07 16:12:07,623] {spark_submit.py:523} INFO - at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
[2021-12-07 16:12:07,624] {spark_submit.py:523} INFO - at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
[2021-12-07 16:12:07,625] {spark_submit.py:523} INFO - at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
[2021-12-07 16:12:07,626] {spark_submit.py:523} INFO - at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
[2021-12-07 16:12:07,626] {spark_submit.py:523} INFO - at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[2021-12-07 16:12:07,627] {spark_submit.py:523} INFO - at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[2021-12-07 16:12:07,628] {spark_submit.py:523} INFO - ... 1 more
[2021-12-07 16:12:27,570] {spark_submit.py:523} INFO - 21/12/07 16:12:27 ERROR StandaloneSchedulerBackend: Application has been killed. Reason: All masters are unresponsive! Giving up.
[2021-12-07 16:12:27,572] {spark_submit.py:523} INFO - 21/12/07 16:12:27 WARN StandaloneSchedulerBackend: Application ID is not initialized yet.
[2021-12-07 16:12:27,584] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 35529.
[2021-12-07 16:12:27,585] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO NettyBlockTransferService: Server created on 7a06f7031264:35529
[2021-12-07 16:12:27,588] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2021-12-07 16:12:27,598] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO SparkUI: Stopped Spark web UI at http://7a06f7031264:4041
[2021-12-07 16:12:27,604] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 7a06f7031264, 35529, None)
[2021-12-07 16:12:27,611] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO StandaloneSchedulerBackend: Shutting down all executors
[2021-12-07 16:12:27,612] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO BlockManagerMasterEndpoint: Registering block manager 7a06f7031264:35529 with 434.4 MiB RAM, BlockManagerId(driver, 7a06f7031264, 35529, None)
[2021-12-07 16:12:27,616] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 7a06f7031264, 35529, None)
[2021-12-07 16:12:27,619] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 7a06f7031264, 35529, None)
[2021-12-07 16:12:27,620] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Asking each executor to shut down
[2021-12-07 16:12:27,629] {spark_submit.py:523} INFO - 21/12/07 16:12:27 WARN StandaloneAppClient$ClientEndpoint: Drop UnregisterApplication(null) because has not yet connected to master
[2021-12-07 16:12:27,657] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2021-12-07 16:12:27,684] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO MemoryStore: MemoryStore cleared
[2021-12-07 16:12:27,686] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO BlockManager: BlockManager stopped
[2021-12-07 16:12:27,700] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO BlockManagerMaster: BlockManagerMaster stopped
[2021-12-07 16:12:27,704] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2021-12-07 16:12:27,720] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO SparkContext: Successfully stopped SparkContext
[2021-12-07 16:12:27,873] {spark_submit.py:523} INFO - 21/12/07 16:12:27 ERROR SparkContext: Error initializing SparkContext.
[2021-12-07 16:12:27,874] {spark_submit.py:523} INFO - java.lang.IllegalArgumentException: requirement failed: Can only call getServletHandlers on a running MetricsSystem
[2021-12-07 16:12:27,876] {spark_submit.py:523} INFO - at scala.Predef$.require(Predef.scala:281)
[2021-12-07 16:12:27,877] {spark_submit.py:523} INFO - at org.apache.spark.metrics.MetricsSystem.getServletHandlers(MetricsSystem.scala:89)
[2021-12-07 16:12:27,878] {spark_submit.py:523} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:603)
[2021-12-07 16:12:27,879] {spark_submit.py:523} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2021-12-07 16:12:27,880] {spark_submit.py:523} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2021-12-07 16:12:27,880] {spark_submit.py:523} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[2021-12-07 16:12:27,882] {spark_submit.py:523} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2021-12-07 16:12:27,882] {spark_submit.py:523} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
[2021-12-07 16:12:27,883] {spark_submit.py:523} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2021-12-07 16:12:27,885] {spark_submit.py:523} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2021-12-07 16:12:27,886] {spark_submit.py:523} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2021-12-07 16:12:27,887] {spark_submit.py:523} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2021-12-07 16:12:27,888] {spark_submit.py:523} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2021-12-07 16:12:27,888] {spark_submit.py:523} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2021-12-07 16:12:27,889] {spark_submit.py:523} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2021-12-07 16:12:27,890] {spark_submit.py:523} INFO - at java.base/java.lang.Thread.run(Thread.java:829)
[2021-12-07 16:12:27,891] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO SparkContext: SparkContext already stopped.
[2021-12-07 16:12:27,892] {spark_submit.py:523} INFO - Traceback (most recent call last):
[2021-12-07 16:12:27,893] {spark_submit.py:523} INFO - File "/opt/***/jobs/persist_db.py", line 52, in <module>
[2021-12-07 16:12:27,894] {spark_submit.py:523} INFO - .config("spark.driver.bindAddress", "localhost") \
[2021-12-07 16:12:27,894] {spark_submit.py:523} INFO - File "/home/***/.local/lib/python3.6/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/session.py", line 228, in getOrCreate
[2021-12-07 16:12:27,895] {spark_submit.py:523} INFO - File "/home/***/.local/lib/python3.6/site-packages/pyspark/python/lib/pyspark.zip/pyspark/context.py", line 392, in getOrCreate
[2021-12-07 16:12:27,896] {spark_submit.py:523} INFO - File "/home/***/.local/lib/python3.6/site-packages/pyspark/python/lib/pyspark.zip/pyspark/context.py", line 147, in __init__
[2021-12-07 16:12:27,897] {spark_submit.py:523} INFO - File "/home/***/.local/lib/python3.6/site-packages/pyspark/python/lib/pyspark.zip/pyspark/context.py", line 209, in _do_init
[2021-12-07 16:12:27,898] {spark_submit.py:523} INFO - File "/home/***/.local/lib/python3.6/site-packages/pyspark/python/lib/pyspark.zip/pyspark/context.py", line 329, in _initialize_context
[2021-12-07 16:12:27,899] {spark_submit.py:523} INFO - File "/home/***/.local/lib/python3.6/site-packages/pyspark/python/lib/py4j-0.10.9.2-src.zip/py4j/java_gateway.py", line 1574, in __call__
[2021-12-07 16:12:27,899] {spark_submit.py:523} INFO - File "/home/***/.local/lib/python3.6/site-packages/pyspark/python/lib/py4j-0.10.9.2-src.zip/py4j/protocol.py", line 328, in get_return_value
[2021-12-07 16:12:27,903] {spark_submit.py:523} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling None.org.apache.spark.api.java.JavaSparkContext.
[2021-12-07 16:12:27,904] {spark_submit.py:523} INFO - : java.lang.IllegalArgumentException: requirement failed: Can only call getServletHandlers on a running MetricsSystem
[2021-12-07 16:12:27,907] {spark_submit.py:523} INFO - at scala.Predef$.require(Predef.scala:281)
[2021-12-07 16:12:27,908] {spark_submit.py:523} INFO - at org.apache.spark.metrics.MetricsSystem.getServletHandlers(MetricsSystem.scala:89)
[2021-12-07 16:12:27,909] {spark_submit.py:523} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:603)
[2021-12-07 16:12:27,910] {spark_submit.py:523} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2021-12-07 16:12:27,911] {spark_submit.py:523} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2021-12-07 16:12:27,912] {spark_submit.py:523} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[2021-12-07 16:12:27,912] {spark_submit.py:523} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2021-12-07 16:12:27,913] {spark_submit.py:523} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
[2021-12-07 16:12:27,914] {spark_submit.py:523} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2021-12-07 16:12:27,915] {spark_submit.py:523} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2021-12-07 16:12:27,915] {spark_submit.py:523} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2021-12-07 16:12:27,916] {spark_submit.py:523} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2021-12-07 16:12:27,918] {spark_submit.py:523} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2021-12-07 16:12:27,918] {spark_submit.py:523} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2021-12-07 16:12:27,919] {spark_submit.py:523} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2021-12-07 16:12:27,920] {spark_submit.py:523} INFO - at java.base/java.lang.Thread.run(Thread.java:829)
[2021-12-07 16:12:27,921] {spark_submit.py:523} INFO - 
[2021-12-07 16:12:27,994] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO ShutdownHookManager: Shutdown hook called
[2021-12-07 16:12:27,995] {spark_submit.py:523} INFO - 21/12/07 16:12:27 INFO ShutdownHookManager: Deleting directory /tmp/spark-7da8021f-2719-45db-a1cf-f1024dde49e6
[2021-12-07 16:12:28,000] {spark_submit.py:523} INFO - 21/12/07 16:12:28 INFO ShutdownHookManager: Deleting directory /tmp/spark-0ffd07db-0515-4fca-ac2a-6202847d1f1b
[2021-12-07 16:12:28,129] {taskinstance.py:1703} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1332, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1458, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1514, in _execute_task
    result = execute_callable(context=context)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 179, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 452, in submit
    self._mask_cmd(spark_submit_cmd), returncode
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --conf spark.master=spark://spark:7077 --name persist data --verbose /opt/***/jobs/persist_db.py laois. Error code is: 1.
[2021-12-07 16:12:28,139] {taskinstance.py:1280} INFO - Marking task as UP_FOR_RETRY. dag_id=scraping_daft_laois, task_id=persist_data, execution_date=20211206T024500, start_date=20211207T161119, end_date=20211207T161228
[2021-12-07 16:12:28,164] {standard_task_runner.py:91} ERROR - Failed to execute job 320 for task persist_data
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/task/task_runner/standard_task_runner.py", line 85, in _start_by_fork
    args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/cli/cli_parser.py", line 48, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/utils/cli.py", line 92, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/cli/commands/task_command.py", line 292, in task_run
    _run_task_by_selected_method(args, dag, ti)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/cli/commands/task_command.py", line 107, in _run_task_by_selected_method
    _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/cli/commands/task_command.py", line 184, in _run_raw_task
    error_file=args.error_file,
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/utils/session.py", line 70, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1332, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1458, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1514, in _execute_task
    result = execute_callable(context=context)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 179, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 452, in submit
    self._mask_cmd(spark_submit_cmd), returncode
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --conf spark.master=spark://spark:7077 --name persist data --verbose /opt/***/jobs/persist_db.py laois. Error code is: 1.
[2021-12-07 16:12:28,212] {local_task_job.py:154} INFO - Task exited with return code 1
[2021-12-07 16:12:28,317] {local_task_job.py:264} INFO - 0 downstream tasks scheduled from follow-on schedule check
